mrmustard/lab_dev/states/base.py

```diff
@@ -248,11 +252,36 @@ def fock_array(self, shape: Optional[Union[int, Sequence[int]]] = None) -> Compl
         """
         return to_fock(self.representation, shape).array
 
-    def phase_space(self) -> tuple[ComplexMatrix, ComplexVector]:
+    def phase_space(
+        self, s: float, characteristic: bool
+    ) -> tuple[complex, ComplexMatrix, ComplexVector]:
         r"""
-        The covariance matrix and the vector of means that describe this state in phase space.
+        The s-parametrized information about the state that describe this state in phase space.
+
+        Args:
+            s: The parameter of the phase space, which corresponds to the measure of the displacement gate :math:`D_s(\gamma) = e^{\frac{s}{2}|\gamma|^2}D(\gamma)`. For example, :math:`s=0`, the information is related to the Wigner distribution and we have the covariance matrix and means vector of the state.
+            characteristic: Whether the phase space is in the characteristic form or not.
+
+            Returns:
+                The coefficient, the covariance matrix and the mean vector of the state in s-parametrized phase space.
         """
-        raise NotImplementedError
+        if not isinstance(self.representation, Bargmann):
+            raise ValueError(f"Can not calculate phase space for ``{self.name}`` object.")
+
+        new_state = self >> _DsMap(self.modes, s=s)  # pylint: disable=protected-access
+        if characteristic:
+            return [
+                Abc_to_cov_mean_for_state_in_characteristic(
+                    new_state.representation.ansatz.A[i],
+                    new_state.representation.ansatz.b[i],
+                    new_state.representation.ansatz.c[i],
+                )
+                for i in np.arange(new_state.representation.ansatz.A.shape[0])
+            ]
+
+        raise NotImplementedError(
+            f"Can not calculate phase space for chracteristic function is ``{characteristic}``."
+        )
```

Comment:
```suggestion
        return [
            Abc_to_cov_mean_for_state_in_characteristic(
                new_state.representation.ansatz.A[i],
                new_state.representation.ansatz.b[i],
                new_state.representation.ansatz.c[i],
            )
            for i in np.arange(new_state.representation.ansatz.A.shape[0])
        ]
```

Reply:
ideally `Abc_to_cov_mean_for_state_in_characteristic` would take the batched `A,b,c`. Also, let's find a shorter name for that function.